File: src/Holefeeder.Ui.Common/Models/Category.cs
namespace Holefeeder.Ui.Common.Models;

public class Category
{
    public Guid Id { get; init; }
    public required string Name { get; init; }
    public string Color { get; init; } = string.Empty;
    public decimal BudgetAmount { get; init; }
    public bool Favorite { get; init; }
}

File: src/Holefeeder.Ui/lib/core/models/category.dart
import 'package:decimal/decimal.dart';

class Category {
  final String id;
  final String name;
  final String color;
  final Decimal budgetAmount;
  final bool favorite;

  const Category({
    required this.id,
    required this.name,
    required this.color,
    required this.budgetAmount,
    required this.favorite,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'] as String,
      name: json['name'] as String,
      color: json['color'] as String,
      budgetAmount: Decimal.parse(json['budgetAmount'].toString()),
      favorite: json['favorite'] as bool,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'color': color,
      'budgetAmount': budgetAmount.toString(),
      'favorite': favorite,
    };
  }
}
